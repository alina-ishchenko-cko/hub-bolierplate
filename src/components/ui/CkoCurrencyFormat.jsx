// @flow
import * as React from 'react';
import { formatNumber } from 'utils/ui.util';
import { WrapStyled } from './styled/CkoCurrencyFormat.sc';

type Props = {
  prefix?: string | React.Node,
  suffix?: string | React.Node,
  value: number,
  currencyName?: string,
  shorten?: boolean,
};

export default class CkoCurrencyFormat extends React.Component<Props> {
  render() {
    return (
      <WrapStyled className="cko-currency-format">
        {this.props.prefix && (
          <span className="prefix">{this.props.prefix}&nbsp;</span>
        )}
        {formatNumber(
          this.props.value,
          this.props.currencyName,
          this.props.shorten
        )}
        {this.props.suffix && (
          <span className="suffix">&nbsp;{this.props.suffix}</span>
        )}
      </WrapStyled>
    );
  }
}
