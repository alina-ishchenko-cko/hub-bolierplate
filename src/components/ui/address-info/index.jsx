// @flow
import React from 'react';

type Props = {
  address: {
    addressLine1: ?string,
    addressLine2: ?string,
    city: ?string,
    postcode: ?string,
    state: ?string,
    country: ?string,
  },
};

export default class AddressInfo extends React.PureComponent<Props> {
  static defaultProps = {
    address: {},
  };

  render() {
    const {
      addressLine1,
      addressLine2,
      city,
      postcode,
      state,
      country,
    } = this.props.address;

    return (
      <div className="address">
        <div>{addressLine1 || ''}</div>
        <div>{addressLine2 || ''}</div>
        <div>{`${city || ''}, ${postcode || ''}`}</div>
        <div>{state || ''}</div>
        <div>{country || ''}</div>
      </div>
    );
  }
}
