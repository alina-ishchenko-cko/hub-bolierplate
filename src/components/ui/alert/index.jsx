// @flow
import * as React from 'react';
import Alert from 'antd/lib/alert';
import { AlertStyled } from './styled/CkoAlert.sc';

type Props = {
  banner?: boolean,
  closable?: boolean,
  type?: string,
  showIcon?: boolean,
  onClose?: Function,
  closeText?: string | React.Node,
  description: string | React.Node,
  message: string | React.Node,
};

export default class CkoAlert extends React.PureComponent<Props> {
  static defaultProps = {
    type: 'info',
  };

  render() {
    return (
      <AlertStyled className="cko-alert">
        <Alert {...this.props} />
      </AlertStyled>
    );
  }
}
