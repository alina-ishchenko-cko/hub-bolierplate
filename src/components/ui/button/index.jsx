// @flow
import * as React from 'react';
import { isUndefined } from 'utils';
import CkoIcon from 'components/ui/icon/';
import { ButtonStyled } from './styled/CkoButton.sc';

type Props = {
  type?: string,
  dataIndex?: any,
  size?: string,
  link?: string,
  icon?: string,
  state?: string,
  disabled?: boolean,
  className?: string,
  block?: boolean,
  value: string,
  children?: React.Node,
  onClick: Function,
  target?: string,
};

export default class CkoButton extends React.PureComponent<Props> {
  static defaultProps = {
    onClick: () => {},
  };

  handleClick = (e: SyntheticEvent<any>) => {
    e.preventDefault();
    e.stopPropagation();

    this.props.onClick(this.props.dataIndex);
  };

  goToLink = () => {
    window.open(this.props.link, this.props.target || '_blank');
  };

  render() {
    const { value, children, icon } = this.props;
    const btnProps = { ...this.props };
    btnProps.className = `cko-button ${this.props.className || ''}`;

    if (!isUndefined(this.props.link)) {
      btnProps.onClick = this.goToLink;
    } else if (!isUndefined(this.props.dataIndex) && !this.props.disabled) {
      btnProps.onClick = this.handleClick;
    }

    return (
      <ButtonStyled {...btnProps}>
        {icon && <CkoIcon name={icon} />}
        {value && <span className="value">{value}</span>}
        {children && children}
      </ButtonStyled>
    );
  }
}
