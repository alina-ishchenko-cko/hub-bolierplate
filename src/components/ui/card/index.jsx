// @flow
import * as React from 'react';
import CkoLoading from 'components/ui/loading/';
import CkoSwitch from 'components/ui/switch/';
import {
  CardWrapStyled,
  ValueWrapStyled,
  CardValueStyled,
  CurrencyStyled,
  TitleStyled,
  SubTitleStyled,
} from './styled/CkoCard.sc';

type Props = {
  loading: boolean,
  title: string,
  subTitle?: string,
  currency?: string,
  value: string,
  switchBtn: {
    btnLabel: Array<string>,
    defaultChecked: boolean,
    onToggleBtn: Function,
  },
  id: string,
  className: string,
};

export default class CkoCard extends React.Component<Props> {
  render() {
    const { switchBtn, title, subTitle, currency, loading } = this.props;
    const props = {};
    if (this.props.id) props.id = this.props.id;
    if (this.props.className) props.className = this.props.className;

    return (
      <CardWrapStyled {...props}>
        {/* Loading */}
        {loading && <CkoLoading circle />}

        {/* Values eg. 2000 eur */}
        {!loading && (
          <ValueWrapStyled switchBtn={switchBtn !== void 0}>
            <CardValueStyled className="amount">
              {this.props.value}
            </CardValueStyled>
            {currency && (
              <CurrencyStyled className="currency">{currency}</CurrencyStyled>
            )}
          </ValueWrapStyled>
        )}

        {/* Switch button */}
        {!loading && switchBtn && <CkoSwitch {...switchBtn} />}
        <div className="clear" />

        {/* Title and Sub-title */}
        {!loading &&
          title && <TitleStyled className="title">{title}</TitleStyled>}
        {!loading &&
          subTitle && (
            <SubTitleStyled className="sub-title">{subTitle}</SubTitleStyled>
          )}
      </CardWrapStyled>
    );
  }
}
