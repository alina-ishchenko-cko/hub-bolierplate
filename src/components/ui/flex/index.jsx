import * as React from 'react';
import styled, { css } from 'react-emotion';

/**
 * Flex Column
 */
const Column = styled('div')`
  display: flex;
  flex-direction: column;
  line-height: 1;
  flex-shrink: ${props => props.shrink || 0};
  flex-basis: ${props => (!isNaN(props.basis) ? props.basis : 'auto')};
  flex-grow: ${props => (!isNaN(props.grow) ? props.grow : 1)};
  margin: ${props => props.margin || 0};
  padding: ${props => props.padding || 0};
  flex-wrap: ${props => props.wrap || 'nowrap'};
  justify-content: ${props => props.justify || 'flex-start'};
  align-items: ${props => props.alignItems || 'stretch'};
  align-content: ${props => props.alignContent || 'stretch'};
  width: ${props => props.width || 'auto'};
`;
export const FlexColumn = props => (
  <Column {...props} className={`cko-flex-column ${props.className || ''}`}>
    {props.children}
  </Column>
);

/**
 * Flex Row
 */
const Row = styled('div')`
  display: flex;
  flex-direction: row;
  line-height: 1;
  flex-shrink: ${props => props.shrink || 0};
  flex-basis: ${props => (!isNaN(props.basis) ? props.basis : 'auto')};
  flex-grow: ${props => (!isNaN(props.grow) ? props.grow : 1)};
  margin: ${props => props.margin || 0};
  padding: ${props => props.padding || 0};
  flex-wrap: ${props => props.wrap || 'nowrap'};
  justify-content: ${props => props.justify || 'flex-start'};
  align-items: ${props => props.alignItems || 'stretch'};
  align-content: ${props => props.alignContent || 'stretch'};
  width: ${props => props.width || 'auto'};
`;
export const FlexRow = props => (
  <Row {...props} className={`cko-flex-row ${props.className || ''}`}>
    {props.children}
  </Row>
);

/**
 * Flex Item
 */
const Item = styled('div')`
  ${flexItemBuilder};
`;

function flexItemBuilder(props) {
  if (props.clear) {
    return css`
      margin: ${props.margin || '0 0 20px 0'};
    `;
  }

  return css`
    flex-basis: ${props.basis || 0};
    flex-grow: ${!isNaN(props.grow) ? props.grow : 1};
    margin: ${props.margin || '0'};
    padding: ${props.padding || 0};
  `;
}
export const FlexItem = props => (
  <Item {...props} className={`cko-flex-item ${props.className || ''}`}>
    {props.children}
  </Item>
);
