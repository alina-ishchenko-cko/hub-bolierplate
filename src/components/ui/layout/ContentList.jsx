// @flow
import * as React from 'react';
import {
  WrapStyled,
  TitleStyled,
  ContentStyled,
} from './styled/ContentList.sc';
type Props = {
  title?: string,
  className?: string,
  children: React.Node,
};

export default class ContentList extends React.Component<Props> {
  render() {
    return (
      <WrapStyled className={`content-wrap ${this.props.className || ''}`}>
        {this.props.title && (
          <TitleStyled>
            <p>{this.props.title}</p>
          </TitleStyled>
        )}
        <ContentStyled>{this.props.children}</ContentStyled>
      </WrapStyled>
    );
  }
}
