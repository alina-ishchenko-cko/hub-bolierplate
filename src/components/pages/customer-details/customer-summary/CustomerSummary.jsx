// @ flow
import * as React from 'react';
import { FlexRow } from 'components/ui/flex';
import ContentList from 'components/ui/layout/ContentList';
import CustomerDetailsValueSummary from '../customer-details-value-summary';
import { formatNumber } from 'utils/ui.util';

type Props = {
  ltv: number,
  orderHistory: number,
  currencyName: string,
};

export default class CustomerSummary extends React.Component<Props> {
  render() {
    return (
      <ContentList title="Summary">
        <FlexRow>
          <CustomerDetailsValueSummary
            value={formatNumber(this.props.ltv || 0, this.props.currencyName)}
            units={this.props.currencyName}
            title="Life time value"
            description="Sum of captured transactions"
          />
          <CustomerDetailsValueSummary
            value={this.props.orderHistory}
            title="Order history"
            description="Number of orders placed so far"
          />
        </FlexRow>
      </ContentList>
    );
  }
}
