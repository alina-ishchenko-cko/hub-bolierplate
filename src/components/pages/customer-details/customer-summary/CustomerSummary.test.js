import React from 'react';
import CustomerSummary from './CustomerSummary';
import ContentList from 'components/ui/layout/ContentList';
import { FlexRow } from 'components/ui/flex';
import * as cs from 'services/currency/currencyService';
import CustomerDetailsValueSummary from '../customer-details-value-summary/';

describe('Customer Summary', () => {
  let wrapper;
  let mockProps;

  beforeAll(() => {
    const mockCurrencies = [
      {
        currencyId: 1,
        name: 'AED',
      },
      {
        currencyId: 2,
        name: 'GBP',
      },
      {
        currencyId: 3,
        name: 'USD',
      },
    ];
    cs.setCurrency(mockCurrencies, 1);

    mockProps = {
      ltv: '5',
      orderHistory: 6,
      currencyName: 'USD',
    };
    wrapper = shallow(<CustomerSummary {...mockProps} />);
  });

  it('renders without crashing', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<CustomerSummary {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains all required components', () => {
    expect(wrapper.find(CustomerDetailsValueSummary)).toHaveLength(2);
    expect(wrapper.find(FlexRow)).toHaveLength(1);
    expect(wrapper.find(ContentList)).toHaveLength(1);
  });

  it('contains summary component for lvt', () => {
    expect(
      wrapper
        .find(CustomerDetailsValueSummary)
        .at(0)
        .props().value
    ).toBe(mockProps.ltv);
  });

  it('contains summary component for order history', () => {
    expect(
      wrapper
        .find(CustomerDetailsValueSummary)
        .at(1)
        .props().value
    ).toBe(mockProps.orderHistory);
  });
});
