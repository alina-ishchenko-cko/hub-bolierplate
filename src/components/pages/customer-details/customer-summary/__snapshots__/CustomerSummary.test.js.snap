// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Summary renders correctly 1`] = `
.css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  line-height: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: auto;
  background-color: #ffffff;
  box-shadow: 0 1px 3px 0 rgba(121,131,149,0.2);
  border-radius: 2px;
  margin-bottom: 10px;
}

.css-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  line-height: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: auto;
  position: relative;
  padding: 20px;
  color: #1e242e;
  font-size: 1.1538461538461537em;
  box-shadow: 0 1px 3px 0 rgba(121,131,149,0.2);
}

.css-2 {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  padding: 25px 30px;
  color: #464f60;
}

.css-2 label {
  color: #798395;
  font-size: 1em;
  margin-bottom: 10px;
  font-weight: 400;
}

.css-2 p {
  font-size: 1.0769230769230769em;
}

.css-2 address {
  line-height: 1.5;
  font-style: normal;
}

.css-2 .ocr-font {
  color: #464f60;
  font-family: 'OCR A Extended';
}

.css-2 .cko-flag {
  border-radius: 3px;
  box-shadow: 0 2px 5px 0 rgba(121,131,149,0.2);
  width: 17px;
  overflow: hidden;
  display: inline-block;
  margin-right: 5px;
}

.css-2 .cko-scheme {
  width: 26px;
  margin-right: 5px;
}

.css-2 .text-capitalize {
  text-transform: capitalize;
}

.css-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  line-height: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: auto;
}

.css-4 {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  margin-right: 25px;
}

.css-4 .summary-value {
  line-height: 1;
  color: #366BCF;
  font-size: 3.076923076923077em;
  margin-bottom: 5px;
  margin-right: 5px;
}

.css-4 .summary-value-title {
  color: #798395;
  font-weight: 500;
  margin-bottom: 5px;
}

.css-4 .summary-value-description {
  color: #ADB0B8;
  white-space: nowrap;
}

.css-4 .summary-value-units {
  font-size: 1.2307692307692308em;
  color: #798395;
}

.css-4 .summary-value-title-container {
  margin-top: 15px;
}

<div
  className="cko-flex-column content-wrap  css-0"
>
  <div
    className="cko-flex-row css-1"
  >
    <p>
      Summary
    </p>
  </div>
  <div
    className="cko-flex-item css-2"
  >
    <div
      className="cko-flex-row  css-3"
    >
      <div
        className="cko-flex-item css-4"
        size={undefined}
      >
        <span
          className="summary-value"
        >
          5
        </span>
        <span
          className="summary-value-units"
        >
          USD
        </span>
        <div
          className="summary-value-title-container"
        >
          <div
            className="summary-value-title"
          >
            Life time value
          </div>
          <div
            className="summary-value-description"
          >
            Sum of captured transactions
          </div>
        </div>
      </div>
      <div
        className="cko-flex-item css-4"
        size={undefined}
      >
        <span
          className="summary-value"
        >
          6
        </span>
        <span
          className="summary-value-units"
        />
        <div
          className="summary-value-title-container"
        >
          <div
            className="summary-value-title"
          >
            Order history
          </div>
          <div
            className="summary-value-description"
          >
            Number of orders placed so far
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
